pkg/filestore/writer_test.go

```diff
@@ -311,10 +385,30 @@ func Test_WriteKeypair(t *testing.T) {
 
 			testBundle := test.testBundle
 			err = w.WriteKeypair(test.meta, testBundle.pk, testBundle.certPEM, testBundle.caPEM)
-			assert.Equal(t, test.expErr, err != nil, "%v", err)
+
+			// make sure to return here if error was expected and received
+			// validation will prevent files from being written
+			if test.expErr {
+				if err == nil {
+					t.Errorf("expected error, got nil")
+				} else {
+					return
+				}
+			} else {
+				assert.NoError(t, err)
+			}
 
 			files, err := store.ReadFiles("vol-id")
 			assert.NoError(t, err)
+
+			// there is some entropy in the PKCS12 process, so we cannot predict the contents of the file
+			if test.isPKCS12 {
+				key := test.meta.VolumeContext["csi.cert-manager.io/keystore-file"]
+				assert.NotEmpty(t, files[key])
+
+				// we'll delete the key to let the assertion for expFiles proceed
+				delete(files, key)
+			}
```

Comment:
Can we not at least ensure when we decode the PKCS12 file that we get the contents that we expect?

Reply:
We are unit test the `WriteKeyPair`-function here, not its dependencies.

The testing you describe is already tested in `pkcs12_test.go`, I don't think it needs be tested here also.